tests/ecosystem/upgrade/test_upgrade.py

```diff
@@ -1,66 +1,31 @@
 import logging
 
+import pytest
+
 from ocs_ci.framework.testlib import ocs_upgrade
-from ocs_ci.framework import config
-from ocs_ci.ocs.cluster import CephCluster, CephHealthMonitor
-from ocs_ci.ocs.resources.storage_cluster import (
-    ocs_install_verification
-)
-from ocs_ci.utility.utils import TimeoutSampler
-from ocs_ci.ocs.ocs_upgrade import (
-    verify_image_versions,
-    OCSUpgrade,
-)
-from ocs_ci.ocs.exceptions import TimeoutException
+from ocs_ci.ocs.ocs_upgrade import run_ocs_upgrade, tier4_functions_to_run
+
 
 log = logging.getLogger(__name__)
 
 
+@pytest.mark.polarion_id("OCS-1579")
```

Comment:
consider using pytest.parametrize so you do something like:


```python

@pytest.mark.parametrize(
    argnames=["operation"], argvalues=[
        pytest.param(
            "worker_abrupt_shutdown", marks=pytest.mark.polarion_id("OCS-1579")
        ),
        pytest.param(
            "worker_perm_shutdown", marks=pytest.mark.polarion_id("OCS-1575")
        ),
    ]    
)
def test_worker_node_abrupt_shutdown(nodes):
    run_ocs_upgrade(nodes, operation)

```

Then, you will not need https://github.com/red-hat-storage/ocs-ci/pull/2453/files#diff-1ba036d13bfcf92845199eee581b5e65R451-R462

Reply:
fixed

Reply:
after discussion with @petr-balogh, i will split it again into 2 functions as it will be easier to execute and track results